kamu-adapter-http/src/http_server_simple_transfer_protocol.rs

```diff
@@ -162,15 +162,10 @@ pub async fn dataset_push_ws_upgrade_handler(
 pub async fn dataset_pull_ws_upgrade_handler(
     ws: axum::extract::ws::WebSocketUpgrade,
     dataset: Extension<Arc<dyn Dataset>>,
-    axum::TypedHeader(api_host): axum::TypedHeader<axum::headers::Host>,
-    axum::extract::Path(params): axum::extract::Path<HashMap<String, String>>,
+    host: axum::extract::Host,
```

Comment:
Notice the use of different `Host` extractor - the one you were using did not respect the "X-Forwarded-Host" or other types of proxy headers.